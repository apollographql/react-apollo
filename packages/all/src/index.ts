// Common + hooks
export * from '@apollo/react-common';

// @apollo/react-components
export {
  Query,
  Mutation,
  Subscription,
  QueryComponentOptions,
  MutationComponentOptions,
  SubscriptionComponentOptions
} from '@apollo/react-components';

// @apollo/react-hoc
export {
  graphql,
  withQuery,
  withMutation,
  withSubscription,
  withApollo,
  QueryControls,
  DataValue,
  DataProps,
  MutateProps,
  ChildProps,
  ChildDataProps,
  ChildMutateProps,
  OptionProps,
  OperationOption,
  WithApolloClient
} from '@apollo/react-hoc';

// SSR
export {
  getMarkupFromTree,
  getDataFromTree,
  renderToStringWithData
} from '@apollo/react-ssr';
