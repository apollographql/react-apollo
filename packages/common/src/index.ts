export {
  ApolloContextValue,
  ApolloContextOptions,
  getApolloContext,
  resetApolloContext
} from './context/ApolloContext';
export { ApolloProvider } from './context/ApolloProvider';
export { ApolloConsumer } from './context/ApolloConsumer';

export {
  parser,
  operationName,
  DocumentType,
  IDocumentDefinition
} from './parser/parser';

export * from './types/types';
