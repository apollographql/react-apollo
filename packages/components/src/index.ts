export {
  ApolloProvider,
  ApolloConsumer,
  getApolloContext,
  resetApolloContext
} from '@apollo/react-common';

export { Query } from './Query';
export { Mutation } from './Mutation';
export { Subscription } from './Subscription';

export * from './types';
