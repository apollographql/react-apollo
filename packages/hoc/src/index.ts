export { graphql } from './graphql';

export { withQuery } from './query-hoc';
export { withMutation } from './mutation-hoc';
export { withSubscription } from './subscription-hoc';
export { withApollo } from './withApollo';

export {
  ApolloProvider,
  ApolloConsumer,
  getApolloContext,
  resetApolloContext
} from '@apollo/react-common';

export * from './types';
