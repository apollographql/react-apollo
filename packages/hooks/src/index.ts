export {
  ApolloProvider,
  ApolloConsumer,
  getApolloContext,
  resetApolloContext
} from '@apollo/react-common';

export { useQuery } from './useQuery';
export { useLazyQuery } from './useLazyQuery';
export { useMutation } from './useMutation';
export { useSubscription } from './useSubscription';
export { useApolloClient } from './useApolloClient';

export { RenderPromises } from './ssr/RenderPromises';

export * from './types';
