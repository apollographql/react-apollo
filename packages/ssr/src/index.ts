export { getMarkupFromTree, getDataFromTree } from './getDataFromTree';
export { renderToStringWithData } from './renderToStringWithData';
export * from './types';
