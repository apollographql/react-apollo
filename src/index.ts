export * from './browser';
export * from './server';
