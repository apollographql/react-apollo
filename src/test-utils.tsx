import * as React from 'react';
import ApolloClient from 'apollo-client';
import { DefaultOptions, Resolvers } from 'apollo-client';
import { InMemoryCache as Cache } from 'apollo-cache-inmemory';

import { ApolloProvider } from './index';
import { MockedResponse, MockLink } from './test-links';
import { ApolloCache } from 'apollo-cache';
export * from './test-links';

export interface MockedProviderProps<TSerializedCache = {}> {
  mocks?: ReadonlyArray<MockedResponse>;
  addTypename?: boolean;
  defaultOptions?: DefaultOptions;
  cache?: ApolloCache<TSerializedCache>;
  resolvers?: Resolvers;
  childProps?: object;
}

export interface MockedProviderState {
  client: ApolloClient<any>;
}

export class MockedProvider extends React.Component<MockedProviderProps, MockedProviderState> {
  public static defaultProps: MockedProviderProps = {
    addTypename: true,
  };

  constructor(props: MockedProviderProps) {
    super(props);

    const {
      mocks,
      addTypename,
      defaultOptions,
      cache,
      resolvers,
    } = this.props;
    const client = new ApolloClient({
      cache: cache || new Cache({ addTypename }),
      defaultOptions,
      link: new MockLink(mocks || [], addTypename),
      resolvers,
    });

    this.state = { client };
  }

  public render() {
    const { childProps } = this.props;

    return (
      <ApolloProvider client={this.state.client}>
        {React.cloneElement(React.Children.only(this.props.children), { ...childProps })}
      </ApolloProvider>
    );
  }

  public componentWillUnmount() {
    // Since this.state.client was created in the constructor, it's this
    // MockedProvider's responsibility to terminate it.
    this.state.client.stop();
  }
}
