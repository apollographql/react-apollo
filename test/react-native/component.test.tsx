import { Text } from 'react-native';
import * as React from 'react';
// Note: test renderer must be required after react-native.
import * as renderer from 'react-test-renderer';
import ApolloClient from 'apollo-client';
import { InMemoryCache as Cache } from 'apollo-cache-inmemory';
import gql from 'graphql-tag';
import { ApolloProvider, ChildProps, graphql } from '../../src';
import { mockSingleLink } from '../../src/test-utils';
import '../test-utils/toEqualJson';

describe('App', () => {
  it('renders correctly', () => {
    const query = gql`
      query people {
        allPeople(first: 1) {
          people {
            name
          }
        }
      }
    `;

    interface Data {
      allPeople: { people: { name: string } };
    }
    const link = mockSingleLink({
      request: { query },
      result: { data: { allPeople: { people: { name: 'Luke Skywalker' } } } },
    });
    const client = new ApolloClient({
      link,
      cache: new Cache({ addTypename: false }),
    });

    const ContainerWithData = graphql(
      query,
    )(({ data }: ChildProps<{}, Data>) => {
      if (data.loading) return <Text>Loading...</Text>;
      return <Text>{data.allPeople.people.name}</Text>;
    });
    const output = renderer.create(
      <ApolloProvider client={client}>
        <ContainerWithData />
      </ApolloProvider>,
    );
    expect(output.toJSON()).toMatchSnapshot();
  });

  it('executes a query', done => {
    jest.useRealTimers();
    const query = gql`
      query people {
        allPeople(first: 1) {
          people {
            name
          }
        }
      }
    `;
    interface Data {
      allPeople: { people: { name: string } };
    }
    const data1 = { allPeople: { people: { name: 'Luke Skywalker' } } };
    const link = mockSingleLink({
      request: { query },
      result: { data: data1 },
    });
    const client = new ApolloClient({
      link,
      cache: new Cache({ addTypename: false }),
    });

    class Container extends React.Component<ChildProps<{}, Data>> {
      componentWillReceiveProps(props) {
        expect(props.data.loading).toBeFalsy();
        expect(props.data.allPeople.people.name).toEqualJson(
          data1.allPeople.people.name,
        );
        done();
      }
      render() {
        if (this.props.data.loading) return <Text>Loading...</Text>;
        return <Text>{this.props.data.allPeople.people.name}</Text>;
      }
    }

    const ContainerWithData = graphql(query)(Container);

    renderer.create(
      <ApolloProvider client={client}>
        <ContainerWithData />
      </ApolloProvider>,
    );
  });
});
