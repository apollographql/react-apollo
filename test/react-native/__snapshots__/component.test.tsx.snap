// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Loading...
</Text>
`;
